src/nrtk_explorer/app/embeddings.py

```diff
@@ -177,18 +188,33 @@ def on_move(self, camera_position):
     def set_on_hover(self, fn):
         self._on_hover_fn = fn
 
-    def on_hover(self, point):
-        self.state.highlighted_point = point
-        image_id = -1
-        if point is not None and point in self.state.user_selected_points_indices:
-            image_id = self.state.images_ids[int(point)]
+    def on_hover(self, point_index):
+        self.state.highlighted_point = point_index
+        image_id = ""
+        if point_index is not None:
+            original_image_point_index = point_index
+            if point_index >= len(self.state.points_sources):
+                image_kind = "transformed_img_"
+                original_image_point_index = self.state.user_selected_points_indices[
+                    point_index - len(self.state.points_sources)
+                ]
+            else:
+                image_kind = "img_"
+            dataset_id = self.state.images_ids[original_image_point_index]
+            image_id = f"{image_kind}{dataset_id}"
+
         if self._on_hover_fn:
             self._on_hover_fn(image_id)
 
-    def on_image_hovered(self, id_, is_transformation):
+    def on_image_hovered(self, id_):
+        if id_ == "":
+            self.state.highlighted_point = -1
+            return
```

Comment:
This change is unneeded the original code deals with this in a more structured way

```
      if id_ in self.state.images_ids:
      else:
        self.state.highlighted_point = -1
```

Reply:
True. I updated the function to not short circut like this.